pkg/minibroker/postgres.go

```diff
@@ -33,30 +33,42 @@ func (p PostgresProvider) Bind(services []corev1.Service, params map[string]inte
 	host := buildHostFromService(service)
 
 	database := ""
-	dbVal, ok := params["postgresDatabase"]
+	dbVal, ok := params["postgresqlDatabase"]
 	if ok {
 		database = dbVal.(string)
 	}
 
 	var user, password string
-	userVal, ok := params["postgresUser"]
+	userVal, ok := params["postgresqlUsername"]
 	if ok {
 		user = userVal.(string)
 	} else {
 		user = "postgres"
 	}
 
-	passwordVal, ok := chartSecrets["postgres-password"]
-	if !ok {
-		// Chart versions 2.0+ use postgresqlPassword instead of postresPassword
-		// See https://github.com/kubernetes-sigs/minibroker/issues/17
-		passwordVal, ok = chartSecrets["postgresql-password"]
-
+	if user != "postgres" {
+		// postgresql-postgres-password is used when postgresqlPostgresPassword is set and
+		// postgresqlUsername is not 'postgres'.
+		passwordVal, ok := chartSecrets["postgresql-postgres-password"]
+		if !ok {
+			passwordVal, ok = chartSecrets["postgresql-password"]
+			if !ok {
+				return nil, errors.Errorf("password not found in secret keys")
+			}
+		}
+		password = passwordVal.(string)
+	} else {
+		passwordVal, ok := chartSecrets["postgres-password"]
```

Comment:
When I originally wrote that logic, I checked for the older secret key first "postgres-password" then checked for the newer "postgresql-password" if it wasn't present. 

Now that the majority of the logic has been revamped to match the new chart logic, it may make it easier to read if we check for "postgresql-password" first here, then fallback to the old 1.0 chart key (updating the comment to indicate it's for backwards compat)

Just a suggestion for maintaining this because reading the logic here is becoming convoluted.

Reply:
True! I'll also do the same for the db and user.

Reply:
Created https://github.com/kubernetes-sigs/minibroker/pull/96 to fix this.